### PR TITLE
Migration, WebClient & Lint Fixes

### PR DESCRIPTION
**Pull request by nodinosaur**
1. Migration to `ViewBinding` from `Kotlin Synthetic` (deprecated)
2. Adding a dedicated `WebClient` to help clean up `BrowserActivityNative`
3. Migrate package names to reflect the new App Id (and something more OSS)
4. Update the About info screen
5. Bump `Crashlytics` dependencies (my fork has a google-services.json for dev)
6. A good few Lint error fixes

Happy to huddle if it helps

(please test the app by run the app for a good length of time)

### CODE REVIEW

**Review by TheTimeWalker — COMMENTED** [review]
This looks pretty good! For now I've only looked through the code, mentioning some nitpicks and specifically the OKHTTP bump which would break KitKat devices. I wlil be trying on real devices soon just to be sure everything is stable

File: WallPanelApp/src/main/java/xyz/wallpanel/app/ui/activities/BrowserActivityNative.kt [hunk 1]
```diff
@@ -346,72 +273,45 @@ class BrowserActivityNative : BaseBrowserActivity(), LifecycleObserver {
...
+    override fun askForWebkitPermission(permission: String, requestCode: Int) {
+        if (ContextCompat.checkSelfPermission(
+                applicationContext,
+                permission
+            ) != PackageManager.PERMISSION_GRANTED
+        ) {
             // Should we show an explanation?
             if (ActivityCompat.shouldShowRequestPermissionRationale(this, permission)) {
-                // Show an expanation to the user *asynchronously* -- don't block
+                // Show an expansion to the user *asynchronously* -- don't block
```
[the remark this answers is not in the document]
Comment (nodinosaur) [reply WallPanelApp/src/main/java/xyz/wallpanel/app/ui/activities/BrowserActivityNative.kt]: edited the typo

File: WallPanelApp/src/main/java/xyz/wallpanel/app/network/MQTT5Service.kt [hunk 2]
```diff
@@ -244,39 +247,45 @@ class MQTT5Service(private var context: Context, options: MQTTOptions,
...
+        private const val ONLINE = "online"
+        private const val OFFLINE = "offline"
+        private const val CONNECTION = "connection"
     }
-}
\ No newline at end of file
+}
+
+
+
```
[the remark this answers is not in the document]
Comment (nodinosaur) [reply WallPanelApp/src/main/java/xyz/wallpanel/app/network/MQTT5Service.kt:295]: not sure, but reapplied AS formatting, maybe it has removed?

File: WallPanelApp/build.gradle [hunk 3]
```diff
@@ -166,7 +170,7 @@ dependencies {
 
     // OKHttp
     implementation 'com.squareup.okhttp3:logging-interceptor:3.4.1'
-    implementation 'com.squareup.okhttp3:okhttp:3.12.13'
+    implementation 'com.squareup.okhttp3:okhttp:3.14.9'
```
[the remark this answers is not in the document]
Comment (nodinosaur) [reply WallPanelApp/build.gradle]: reverted the change

**Comment by TheTimeWalker** [comment]
While testing out I came across a crash loop on one of my Oreo devices where after trying to start up the camera but failing, it then gives a NullPointerException at CrashlyticsDebugTree. Haven't gotten into debugging this yet.

```java
Fatal Exception: java.lang.RuntimeException: Unable to create service xyz.wallpanel.app.network.WallPanelService: java.lang.NullPointerException
       at android.app.ActivityThread.handleCreateService(ActivityThread.java:3376)
       at android.app.ActivityThread.-wrap4()
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1701)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:164)
       at android.app.ActivityThread.main(ActivityThread.java:6523)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:857)
```

```java
Caused by java.lang.NullPointerException
       at xyz.wallpanel.app.utils.CrashlyticsDebugTree.log(CrashlyticsDebugTree.kt:17)
       at timber.log.Timber$Tree.prepareLog(Timber.java:532)
       at timber.log.Timber$Tree.e(Timber.java:450)
       at timber.log.Timber$1.e(Timber.java:306)
       at timber.log.Timber.e(Timber.java:83)
       at xyz.wallpanel.app.modules.CameraReader.startCamera(CameraReader.kt:132)
       at xyz.wallpanel.app.network.WallPanelService.configureCamera(WallPanelService.kt:381)
       at xyz.wallpanel.app.network.WallPanelService.onCreate(WallPanelService.kt:177)
       at android.app.ActivityThread.handleCreateService(ActivityThread.java:3366)
       at android.app.ActivityThread.-wrap4()
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1701)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:164)
       at android.app.ActivityThread.main(ActivityThread.java:6523)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:857)
```

**Comment by nodinosaur** [comment]
Understood, but this is not anything to do with the Pull Request above, this is a pre existing issue I believe.

**Comment by nodinosaur** [comment]
Fixed a potential NPE

**Comment by nodinosaur** [comment]
I also see camera errors on the Nexus 9 with SlimRom unfortunately, these JNI bugs are difficult to trace

**Comment by TheTimeWalker** [comment]
Ah, sorry about that. I didn't mean the camera issue specifically. I just meant the CrashlyticsDebugTree.kt as I assumed with the new utils file in this PR, it could potentially also crash with NPE on other exceptions.

**Comment by nodinosaur** [comment]
Yep, that has been addressed with [c0a2f09](https://github.com/TheTimeWalker/wallpanel-android/pull/16/commits/c0a2f09b1281286337b5981a6130097dc28c4e91)

**Comment by TheTimeWalker** [comment]
This looks very good! The only issue left that I have found is that on KitKat, it seems like the WebView does not get bound on the layout properly. There's only a white page unfortunately regardless on what URL I have set. By debugging, I see the pages do load in the background. I assume this has something to do with the new ViewBinding that replaced the deprecated synthetic way. I personally haven't found a solution yet, maybe you have an idea how to solve this 😄 

If that could be solved then this would LGTM

**Comment by nodinosaur** [comment]
This awkward... I don't have any KitKat devices, and I cannot download an image in Android Studio... I am open to suggestions?
Screenshot 2022-05-16 at 19 26 03

**Comment by nodinosaur** [comment]
I just checked the Kitkat scenario against the `TheTimeWalker/wallpanel-android` `master` branch and see exactly the same result. Emulator: Nexus 7 KitKat
kitkat_nexus7

**Comment by TheTimeWalker** [comment]
Oops! Sorry, it seems like I've been confused all along! I've been mentioning the wrong Android version where I found the issue, sorry about that!

The white screen issue mentioned is actually on Lollipop 5.1 and not Kitkat 4.4. Maybe you could also quickly check out if it's running fine on Android 5.1 emulation for you? Because that's a version we definitely will be keeping support for a while now, so it would be good that it doesn't regress there. That's what I see on Lollipop at least with every URL:
image

Just to mention Kitkat, what you see is pretty much how it should be. Because KitKat runs with Chrome v30, it's unfortunately unsupported by Home Assistant. WallPanel can still be used for other dashboards however! :)

**Comment by nodinosaur** [comment]
Lollipop works fine:
Emulator of Nexus 9 Lollipop
Emulator_and_Details

**Comment by TheTimeWalker** [comment]
API 25 is Android 7.1.1 Nougat if the device name is correct 😄

**Comment by nodinosaur** [comment]
Doh! Yes, You are right.
Attached Lollipop, not sure about WebView version, but there is some rendering
Nexus9_API_22

and
Nexus9_API_22_Lollipop5-1


But I tested it with Wallpanel.xyz:


Nexus9_API_22_wallpanel

**Comment by TheTimeWalker** [comment]
Thanks, I guess I'll chalk it up on my setup doing something weird in that regard.
At least locally on devices it seems to work pretty fine as it has before. I haven't noticed anything severe, so this looks good to me!

Thank you very much for this PR and for the work. This definitely helped me quite a lot since quite a lot of things were something I planned to do, but didn't come around yet. I really appreciate the whole cleanup and porting done to modernize the code stack a bit further.

**Comment by starkillerOG** [comment]
@TheTimeWalker could you release a new version?
I would like to test these changes and see if they improve the crashes I experiance.

**Comment by nodinosaur** [comment]
This is more of a `code package refactor` and (in my case) some tweaks to crash logging rather than a  series of fixes.
But please check out the code, drop your `google-services.json` into the `WallPanelApp` directory, compile, deploy it to your test device and see how you get on.

**Pull request merged**